packages/typeorm/README.md

```diff
@@ -0,0 +1,145 @@
+`TypeOrmComponent` component. Import them from `@loopback/typeorm` and configure
+your app as shown below.
+
+```ts
+import {TypeOrmComponent, TypeOrmMixin} from '@loopback/typeorm';
+...
+export class AppaApplication extends BootMixin(
+  ServiceMixin(TypeOrmMixin(RestApplication)),
+) {
+  constructor(options: ApplicationConfig = {}) {
+    super(options);
+    ...
+    this.component(TypeOrmComponent);
+    ...
+  }
+}
+
+```
+
+### Creating Entities
+
+[Entities](https://typeorm.io/#/entities) are equivalent to Juggler's Models.
+Define the entities as usual and keep them in a directory named `entities`, or
+even `models` or any name, just make sure to load them correctly in the
+connections.
+
+```ts
+// src/entities/book.entity.ts
+import {Entity, Column, PrimaryColumn} from 'typeorm';
+@Entity()
+export class Photo {
+  @PrimaryColumn()
+  id: number;
+
+  @Column()
+  title: string;
+
+  @Column()
+  isPublished: boolean;
+}
+```
+
+### Creating Connections
+
+[Connections](https://typeorm.io/#/connection) are equivalent to Juggler
+datasources. They contain the connectivity and other details about the
+databases. Define the connection in files with a `.connection.ts` extension and
+keep them in the `connections` directory of the project.
+
+```ts
+// src/connections/sqlite.connection.ts
+import path from 'path';
+import {SqliteConnectionOptions} from '@loopback/typeorm';
+import {Book} from '../entities/';
+
+export const SQLLITE_CONNECTION: SqliteConnectionOptions = {
+  type: 'sqlite',
+  database: path.join(__dirname, 'sqlite-test-db'),
+  entities: [Book],
+  synchronize: true,
+};
+```
+
+Make sure to install the underlying database driver. For example, if you are
+using SQlite, you'll need to install `sqlite3`.
+
+```sh
+npm install sqlite3
+```
+
+Refer to the
+[TypeORM documentation](https://github.com/typeorm/typeorm#installation) for the
+supported databases and the underlying drivers.
+
+### Creating controllers
+
+Controllers continue to work as usual. And you don't have to create repositories
+since TypeORM creates them for you; just inject them in the controllers. The
+repository API is 100% TypeORM
+[repository API](https://typeorm.io/#/repository-api).
+
+```ts
+// src/controllers/book.controller.ts
+import {get, post, Request, requestBody} from '@loopback/rest';
+import {getModelSchemaRef, Repository, typeorm} from '@loopback/typeorm';
+import {Book} from '../entities';
+
+export class BookController {
+  @typeorm.repository(Book) private bookRepo: Repository<Book>;
+
+  constructor() {}
+
+  // Create a new book
+  @post('/book')
+  async create(@requestBody() data: Book) {
+    const book = new Book();
+    book.title = data.title;
+    book.published = false;
+    return await this.bookRepo.save(book);
+  }
+
+  // Find book by title
+  @get('/note/{title}')
+  async findByTitle(@param.path.string('title') title: string) {
+    return await this.bookRepo.find({title});
+  }
+}
+```
+
```

Comment:
Please add a new section to list known limitations - things/features/use-cases we know about but not implemented yet.

Few candidates:

- Full JSON/OpenAPI schema for models, including variants like with/without id, with/without relations, partial, etc.
- Json/OpenAPI schema to describe the supported `filter` format
- Support for LoopBack-style filters
- Custom repository classes (e.g. to implement `bookRepo.findByTitle(title)`).
- Database migration

Scenarios we may need to verify and document:

- When using multiple connections, how to specify which model is backed by which connection? E.g. `Category` and `Product` is accessed via `db` connection, while `Customer` and `Address` is accessed via `crm` connection.

Reply:
@hacksparrow I think your list of limitations is missing the following items:

- Json/OpenAPI schema to describe the supported filter format
- Support for LoopBack-style filters
- Custom repository classes (e.g. to implement bookRepo.findByTitle(title)).
- When using multiple connections, how to specify which model is backed by which connection? E.g. Category and Product is accessed via `db` connection, while Customer and Address is accessed via `crm` connection.

Reply:
> When using multiple connections, how to specify which model is backed by which connection? E.g. Category and Product is accessed via db connection, while Customer and Address is accessed via crm connection.

Like this:

```ts
export const AssetConnection: ConnectionOptions = {
  name: 'db',
  type: 'sqlite',
  database: 'assets',
  entities: [Category, Product],
  synchronize: true,
};
```

```ts
export CrmConnection: ConnectionOptions = {
  name: 'crm',
  type: 'mysql',
  database: 'users',
  entities: [Customer, Address],
  synchronize: true,
};
```

This is more of a TypeORM documentation thing.

Reply:
> I think your list of limitations is missing the following items:

Updated.